site/app/views/submission/HomeworkView.php

```diff
@@ -197,14 +197,37 @@ public function renderLateDayMessage(LateDays $late_days, Gradeable $gradeable,
             $active_days_charged = max(0, $active_days_late - $extensions);
         }
 
+        // process daylight savings banner
+        // check if we have excessive amounts of late days or if the due date and due date + late days is in different time zones
+        // and if the user can submit without it being too late
+        $due_date = $gradeable->getSubmissionDueDate();
+        $due_date_with_late_days = clone $due_date;
+        $due_date_with_late_days->modify('+' . $late_days_allowed . ' days');
+        $today = new \DateTime();
+
+        // if we are past the due date + late days
+        if ($today > $due_date_with_late_days) {
+            $daylight_message_required = false;
+        }
+        elseif ($late_days_allowed <= 365) {
+            $future_due_date = clone $due_date;
+            $daylight_message_required = false;
+            // check every interval of 100
+            for ($i = 0; $i < intdiv($late_days_allowed, 100) && !$daylight_message_required; $i += 1) {
+                $future_due_date->modify('+100 days');
+                $daylight_message_required = $future_due_date->format('I') !== $due_date->format('I');
```

Comment:
How about we leave some comment here?
```suggestion
                //  The format specifier 'I' is used to extract a single character representing the DST indicator for the date. This will be either a '0' (DST not in effect) or a '1' (DST in effect).
                $daylight_message_required = $future_due_date->format('I') !== $due_date->format('I');
```